### PR TITLE
92 training pipeline

### PR DESCRIPTION
**Pull request by jogong2718**
Finished the training script and awaiting review. 

Additional details:
- The transform_data function can be changed to remove the line that removes null values. This is because testing the interim datasets there are null values in the bgl column but I expect the processed data to not have those
- As we are only predicting the largest meals, the ground truth labels are only 0 and 1 (for the "ANNOUNCE_MEAL"). This can be changed depending on future prediction needs
- \__init\__.py files were added to some directories so the file paths would work

Note: When working with importing the ClusterSegmenter, the documentation for the module that the model is implemnted from the sktime website differed from the actual module name. The website states [sktime.annotation.cluster](https://www.sktime.net/en/stable/api_reference/auto_generated/sktime.annotation.clust.ClusterSegmenter.html) when the actual module is [sktime.annotation.clust](https://github.com/sktime/sktime/blob/main/sktime/annotation/clust.py).

### CODE REVIEW

**Comment by RobotPsychologist** [comment]
Thanks @jogong2718, seems like we still need to figure out why this is failing regarding your import from the config file? @Tony911029

**Comment by Tony911029** [comment]
I think there has something to do with importing a custom package (meal_announcement). I can take a look during our meeting tonight

**Comment by Tony911029** [comment]
@RobotPsychologist Can you let me know when you guys are done with this PR. I can do a final fix to fix the pipeline. THanks

**Comment by jogong2718** [comment]
Sorry for the delay @RobotPsychologist. I just finished configuring the unit tests so I think this issue should be done.

**Comment by RobotPsychologist** [comment]
No worries @jogong2718 thanks for doing this, seems like its still failing one test though. @Tony911029  do you have any thoughts on this?

**Comment by jogong2718** [comment]
@Tony911029 Just added you

**Comment by NathanL15** [comment]
Config in linux server error is probably due to the two "meal_identification" folders. Check if it applies to you @Tony911029 and try both folder directories to see if the error persists. Also, you might need to add another meal identification to the config directory as well once you get config recognized.

**Comment by Tony911029** [comment]
@NathanL15 I don't think it is nested meal_identification's issues because it ran before. I will do some comparison with existing main

File: 0_meal_identification/meal_identification/meal_identification/tests/model_training/test_modeling.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+sample data not found
+mockpaths not found
+"""
+import unittest
+from unittest.mock import patch
+import pandas as pd
+from pathlib import Path
+import sys
+import os
+# sys.path.append(R"C:\Users\Nathan\Documents\GitHub\jo_bg_control\0_meal_identification\meal_identification\meal_identification\modeling")
```
Comment (Tony911029) [inline 0_meal_identification/meal_identification/meal_identification/tests/model_training/test_modeling.py:15]: @NathanL15 Are you able to clean this up? Thanks

**Review by Tony911029 — CHANGES_REQUESTED** [review]

**Comment by Tony911029** [comment]
@NathanL15 Fixed. I think you added some __init__ so the package was imported from the wrong place

**Comment by Tony911029** [comment]
@paulpark6 I think the test you just pushed failed

**Comment by paulpark6** [comment]
Working on the issue, my bad.

**Review by RobotPsychologist — CHANGES_REQUESTED** [review]
Why are all these venv files in the PR? Why is there 876 files? lol!

File: eda/trial_eda_applications/eda_Alyssa_DSouza.ipynb
Comment (Tony911029) [inline eda/trial_eda_applications/eda_Alyssa_DSouza.ipynb:1]: why is eda file touched? This should not be touched anymore

File: venv/bin/Activate.ps1 [hunk 2]
```diff
@@ -0,0 +1,247 @@
+<#
```
Comment (Tony911029) [inline venv/bin/Activate.ps1:1]: Please put venv in .gitignore

File: 0_meal_identification/meal_identification/notebooks/1.01-jg-GMMHMM-implementation.ipynb [hunk 3]
```diff
@@ -34,6 +34,7 @@
    ]
   },
   {
```
Comment (Tony911029) [inline 0_meal_identification/meal_identification/notebooks/1.01-jg-GMMHMM-implementation.ipynb:36]: If the content is identical, please rollback the change

**Review by Tony911029 — CHANGES_REQUESTED** [review]

**Review by jogong2718 — COMMENTED** [review]
I think the "update repo" commit may have caused the 897 file changes. I'm going to see if I can revert these changes.

File: 0_meal_identification/meal_identification/notebooks/1.01-jg-GMMHMM-implementation.ipynb (hunk 3 above)
Comment (jogong2718) [reply 0_meal_identification/meal_identification/notebooks/1.01-jg-GMMHMM-implementation.ipynb]: fixed

File: 0_meal_identification/meal_identification/__init__.py
[the remark this answers is not in the document]
Comment (jogong2718) [reply 0_meal_identification/meal_identification/__init__.py:1]: I'll remove it. Im not sure why it was added I think it was to try and fix the module import issue but that didnt help.

**Comment by jogong2718** [comment]
Ok @RobotPsychologist @Tony911029 . Things should be fixed.

**Review by RobotPsychologist — APPROVED** [review]
I think this should be good I don't see any immediate issues!

**Review by Tony911029 — APPROVED** [review]
